### PR TITLE
Make all pages og:aware

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/3540 by making our `base` template add in the `lang` attribute and og metadata, such that any child _can_ opt out by blanking the `social_metadata` block... although I have no idea why any page would do so.

### CODE REVIEW

File: network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html [hunk 1]
```diff
@@ -7,10 +7,6 @@
   <meta name="ga-identifier" content="UA-87658599-15">
 {% endblock %}
 
-{% block socialMetadata %}
```
Comment (Pomax) [inline network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html]: extends from base, so inherits the metadata code from there

File: network-api/networkapi/mozfest/templates/mozfest/404.html [hunk 2]
```diff
@@ -1,9 +1,5 @@
 {% extends "mozfest/mozfest-base.html" %}
 
-{% block socialMetadata %}
```
Comment (Pomax) [inline network-api/networkapi/mozfest/templates/mozfest/404.html]: mozfest-base extends from base, so inherits the metadata code from there

**Comment by Pomax** [comment]
Interestingly, this seems to break some buyersguide tests...

File: network-api/networkapi/templates/pages/base.html [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
   <head>
     <meta charset="utf-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1">{% block socialMetadata %}{% endblock %}
+    <title>{% block pageTitle %}Mozilla Foundation - {% if page.specifics.seo_title %}{{ page.specifics.seo_title }}{% else %}{{ page.title }}{% endif %}{% endblock %}</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    {% block social_metadata %}
+      <meta property='og:type' content='website'>
+      <meta property='og:locale' content='{{ lang_code|to_opengraph_locale }}'>
+      <meta property='wagtail:language' content='{{ lang_code }}'>
+      {% meta_tags %}
```
Comment (youriwims) [inline network-api/networkapi/templates/pages/base.html]: What's `meta_tags` here?

Comment (Pomax) [reply network-api/networkapi/templates/pages/base.html]: the tag used by the wagtail-metadata app to inject all the additional metadata fields from the `Promote` tab.

Comment (Pomax) [reply network-api/networkapi/templates/pages/base.html]: actually, let me move that into its own block.

File: network-api/networkapi/templates/pages/base.html [hunk 4]
```diff
@@ -1,11 +1,24 @@
+{% load wagtailmetadata_tags i18n localization settings_value wagtailuserbar %}
+
+{% get_current_language as lang_code %}
+
 <!DOCTYPE html>
-<html>{% load settings_value wagtailuserbar %}
+<html lang="{{ lang_code }}">{% load settings_value %}
```
Comment (Pomax) [inline network-api/networkapi/templates/pages/base.html]:
```suggestion
<html lang="{{ lang_code }}">
```

File: network-api/networkapi/templates/pages/base.html (hunk 3 above)
Comment (youriwims) [reply network-api/networkapi/templates/pages/base.html]: Ah ok, that makes sense. Thanks!

File: network-api/networkapi/templates/404.html [hunk 5]
```diff
@@ -1,9 +1,9 @@
-{% extends "pages/base.html" %} {% load wagtailcore_tags wagtailimages_tags %}
```
Comment (Pomax) [inline network-api/networkapi/templates/404.html]: the 404 doesn't use any tags, and so doesn't need to load anything.

File: network-api/networkapi/templates/404.html [hunk 6]
```diff
@@ -1,9 +1,9 @@
...
+{% extends "pages/base.html" %}
 
 {% block pageTitle %}Page Not Found{% endblock %}
 
+{% block wagtail_metadata %}{% endblock %}
```
Comment (Pomax) [inline network-api/networkapi/templates/404.html]: 404 is not managed by wagtail in the slightest, so it shouldn't try to inject wagtail metadata.

**Comment by Pomax** [comment]
@youriwims found it, fixed it, slight code changes so marked for rereview just to be sure.

File: network-api/networkapi/templates/404.html (hunk 6 above)
Comment (alanmoo) [reply network-api/networkapi/templates/404.html]: So shouldn't we do the same thing in mozfest/404?

Comment (Pomax) [reply network-api/networkapi/templates/404.html]: that's a really good point - made sure that's the case

File: network-api/networkapi/mozfest/templates/mozfest/404.html [hunk 7]
```diff
@@ -1,16 +1,13 @@
...
-<meta property='og:locale' content='en_US'>
-{% endblock %}
-
 {% block page_title %}
 Mozilla Festival 2019 - Page Not Found
 {% endblock %}
 
 {% block body_id %}{% endblock %}
 
+{% block wagtail_metadata %}{% endblock %}
```
Comment (Pomax) [inline network-api/networkapi/mozfest/templates/mozfest/404.html]: make sure 404 pages do not try to load wagtail-managed metadata!

File: network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html [hunk 8]
```diff
@@ -8,9 +8,8 @@
 {% endblock %}
 
 {% block social_metadata %}
-    <meta property='og:type' content='website'>
-    <meta property='og:locale' content='en_US'>
-    {% meta_tags %}
+  <meta property='og:type' content='website'>
+  <meta property='og:locale' content='en_US'>
```
Comment (Pomax) [inline network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html]: mozfest is simply en_US

File: network-api/networkapi/templates/404.html [hunk 9]
```diff
@@ -1,7 +1,9 @@
-{% extends "pages/base.html" %} {% load wagtailcore_tags wagtailimages_tags %}
+{% extends "pages/base.html" %}
 
 {% block page_title %}Page Not Found{% endblock %}
 
+{% block wagtail_metadata %}{% endblock %}
```
Comment (Pomax) [inline network-api/networkapi/templates/404.html]: make sure 404 pages do not try to load wagtail-managed metadata!

File: network-api/networkapi/wagtailpages/templates/wagtailpages/homepage.html [hunk 10]
```diff
@@ -3,12 +3,6 @@
 
 {% block body_id %}home{% endblock %}
 
-{% block social_metadata %}
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/homepage.html]: inherited from base now

File: network-api/networkapi/wagtailpages/templates/wagtailpages/modular_page.html [hunk 11]
```diff
@@ -2,12 +2,6 @@
 
 {% load wagtailcore_tags mini_site_tags wagtailmetadata_tags %}
 
-{% block social_metadata %}
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/modular_page.html]: inherited from base now

**Comment by Pomax** [comment]
@alanmoo @youriwims take two.

File: network-api/networkapi/templates/pages/base.html [hunk 12]
```diff
@@ -1,21 +1,40 @@
...
     <meta charset="utf-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1">{% block social_metadata %}{% endblock %}
+    <title>{% block pageTitle %}Mozilla Foundation - {% if page.specifics.seo_title %}{{ page.specifics.seo_title }}{% else %}{{ page.title }}{% endif %}{% endblock %}</title>
+
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta name="csrf-token" content="{{ csrf_token }}">
+
+    {% block social_metadata %}
+      <meta property='og:type' content='website'>
+      <meta property='og:locale' content='{{ lang_code|to_opengraph_locale }}'>
```
Comment (youriwims) [inline network-api/networkapi/templates/pages/base.html]: What's `to_opengraph_locale`? _tried looking it up and got sent to my gitmemory profile...didn't even know that existed_ 😮

File: network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html (hunk 8 above)
Comment (youriwims) [reply network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html]: Why is that if you don't mind me asking? 😬

Comment (alanmoo) [reply network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html]: Yeah, there are localization fields for a bunch of the content on it...why hardcode it to English?

Comment (alanmoo) [reply network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html]: Also, was this just a spacing fix?

File: network-api/networkapi/templates/pages/base.html (hunk 12 above)
Comment (Pomax) [reply network-api/networkapi/templates/pages/base.html]: interesting. It was _supposed_ to be a tag, loaded from localization.py, but that files seems to have disappeared in the PR? O_o

File: network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html (hunk 8 above)
Comment (Pomax) [reply network-api/networkapi/mozfest/templates/mozfest/mozfest-base.html]: It was, although if mozfest is properly localized this needs to simply "not override the social_metadata"

File: network-api/networkapi/templates/pages/base.html (hunk 12 above)
Comment (Pomax) [reply network-api/networkapi/templates/pages/base.html]: that file should now be restored ... so weird

File: network-api/networkapi/wagtailpages/templatetags/localization.py [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+    'es': 'es_ES',
+    'de': 'de_DE',
+    'pl': 'pl_PL',
+    'pt': 'pt_BR',  # our main focus is Brazilian Portuguese
+}
+
+
+# This filter turns Wagtail language codes into OpenGraph locale strings
+@register.filter
+def to_opengraph_locale(value):
```
Comment (youriwims) [inline network-api/networkapi/wagtailpages/templatetags/localization.py]: Niceeeeeee 🌟

**Pull request merged**